### ISSUE TITLE
SilverStripe 3.7.3 DBField::scalarValueOnly on ArrayData

### ISSUE THREAD

**Issue by geekdenz** [symbiote/silverstripe-advancedworkflow#399]
in version 4.1.0, on line 109 we call 
`$item->Assignee = new ArrayData($assigneeVars);`
which in SilverStripe 3.7.3 down the line calls
Object.php:
`if ($dbField && $dbField->scalarValueOnly() && !empty($val) && !is_scalar($val)){`
where $dbField is an instance of ArrayData.

This raises an exception because ArrayData does not have the scalarValueOnly() method.
The exception is
`ERROR [User Error]: Uncaught Exception: Object->__call(): the method 'scalarvalueonly' does not exist on 'ArrayData', or the method is not public.
IN POST /admin/pages/edit/EditForm/318?Locale=en_GB
Line 777 in .../framework/core/Object.php

Source
======
  768: 
  769: 				default :
  770: 					throw new Exception (
  771: 						"Object->__call(): extra method $method is invalid on $class:"
  772: 							. var_export($config, true)
  773: 					);
  774: 			}
  775: 		} else {
  776: 			// Please do not change the exception code number below.
* 777: 			throw new Exception("Object->__call(): the method '$method' does not exist on '$class', or the
       method is not public.", 2175);
  778: 		}
  779: 	}
  780: 
  781: 	//
       --------------------------------------------------------------------------------------------------------------
  782: 
  783: 	/**

Trace
=====
SS_Object->__call(scalarvalueonly,Array)
DataObject.php:2682

DataObject->setField(Assignee,ArrayData)
ViewableData.php:142

ViewableData->__set(Assignee,ArrayData)
ViewableData.php:720

ViewableData_Customised->__set(Assignee,ArrayData)
NotifyUsersWorkflowAction.php:109

NotifyUsersWorkflowAction->execute(WorkflowInstance)
WorkflowInstance.php:285

WorkflowInstance->execute()
WorkflowInstance.php:366

WorkflowInstance->performTransition(WorkflowTransition)
WorkflowInstance.php:301

WorkflowInstance->execute()
WorkflowService.php:248

WorkflowService->startWorkflow(SummaryPage,7)
AdvancedWorkflowExtension.php:29

AdvancedWorkflowExtension->startworkflow(Array,CMSForm,SS_HTTPRequest)
Object.php:746

SS_Object->__call(startworkflow,Array)
Form.php:466

Form->httpSubmission(SS_HTTPRequest)
RequestHandler.php:296

RequestHandler->handleAction(SS_HTTPRequest,httpSubmission)
RequestHandler.php:208

RequestHandler->handleRequest(SS_HTTPRequest,DataModel)
RequestHandler.php:230

RequestHandler->handleRequest(SS_HTTPRequest,DataModel)
Controller.php:151

Controller->handleRequest(SS_HTTPRequest,DataModel)
LeftAndMain.php:474

LeftAndMain->handleRequest(SS_HTTPRequest,DataModel)
AdminRootController.php:92

AdminRootController->handleRequest(SS_HTTPRequest,DataModel)
Director.php:385

Director::handleRequest(SS_HTTPRequest,Session,DataModel)
Director.php:149

Director::direct(/admin/pages/edit/EditForm/318,DataModel)
main.php:211`

I currently do not know how to even approach this problem because the code is rather abstract. Could someone have a quick look and see how we can work around this or better yet fix the issue with a patch or better yet with a minor version upgrade?

**Comment by robbieaverill** [comment]
Closing as a duplicate of https://github.com/symbiote/silverstripe-advancedworkflow/issues/391 (already fixed, but probably not released)

**Comment by geekdenz** [comment]
Thanks @robbieaverill ! I didn't find that bit of information. It seems rather hard to discover. Also, I had problems checking out exactly that version with composer require which we needed for our setup. If you have similar issues in the future anyone, I've created a branch here that I was able to checkout with composer require and a repository in composer.json:

https://github.com/geekdenz/silverstripe-advancedworkflow/tree/landcare
